src/components/TableComponent/index.jsx

```diff
@@ -1,166 +1,120 @@
-import React from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
 
-import { Pagination, Table } from '@edx/paragon';
+import { Pagination, DataTable, Alert } from '@edx/paragon';
+import { Error, WarningFilled } from '@edx/paragon/icons';
 import 'font-awesome/css/font-awesome.css';
 
 import TableLoadingSkeleton from './TableLoadingSkeleton';
-import TableLoadingOverlay from '../TableLoadingOverlay';
-import StatusAlert from '../StatusAlert';
 import { updateUrl } from '../../utils';
 
-class TableComponent extends React.Component {
```

Comment:
Long term, I'm not entirely sure the benefit of keeping `TableComponent`/`TableContainer` now that `DataTable` exists and am considering whether this refactor should move away from these components in favor of implementing `DataTable` directly where needed.

That way, we don't need to refactor/maintain a single table that supports every single possible table use case throughout the application vs. just implementing the bits and pieces of `DataTable` as needed for each individual use case instead.